src/metanetx/data.py

```diff
@@ -128,6 +166,23 @@ def match(self, query):
             or query.lower() in self.name.lower()
         )
 
+    def exact_match(self, query):
+        """
+        Match a string exact, case-insensitively against this metabolite.
+
+        Returns True if the query matches the metanetx identifier, name or any
+        annotation identifier.
+        """
+        query = query.lower()
+        if query == self.mnx_id.lower() or query == self.name.lower():
+            return True
+
+        for identifiers in self.annotation.values():
+            if query in [identifier.lower() for identifier in identifiers]:
+                return True
+
+        return False
```

Comment:
For a larger speedup, you can
```py
# data.py
# dict, keyed by exact tokens
reactions_by_exact = { }

# parser.py
for reaction in reactions:
  reactions_by_exact[reaction.mnx_id.lower()] = reaction
  reactions_by_exact[reaction.name.lower()] = reaction
  for identifiers in self.annotation.values():
    for identifier in identifiers:
      reactions_by_exact[identifier.lower()] = reaction

# resources.py
Batch
  return [
    reactions_by_exact[q.lower()] or None
    for q in query
  ]
```

Reply:
(if there's any overlap between reaction1.mnx_id === reaction2.annotation.x[1], I guess code should be reordered so `reactions_by_exact[mnx_id] = reaction1` has priority)

Reply:
Ah, the custom index! That's a very interesting idea, it should be a lot faster indeed. Let me give it a try in a different PR. I think MNX IDs are fairly unique, it's more likely to have duplicates within the annotated namespaces. But then we might as well leave the order arbitrary.